### PR TITLE
If index segment doesn't have any entry then return length of main segment as result

### PR DESCRIPTION
**Pull request by Bhupender-Y**
**Change log description**  
If index segment doesn't have any entry then return length of main segment as result. 

**Purpose of the change**  

**What the code does**  
It will return length of main segment as result if index segment doesn't have any entry.

**How to verify it**  
(Optional: steps to verify that the changes are effective)

### CODE REVIEW

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java [hunk 1]
```diff
@@ -69,15 +77,42 @@ private void handleIndexAppend(String segmentName) {
              .whenComplete((length, ex) -> {
                  if (ex == null) {
                      log.trace("Index segment append successful for Segment : {} with length {} ", segmentName, length);
+                 } else if (Exceptions.unwrap(ex) instanceof StreamSegmentNotExistsException) {
+                     logWarningMessage(segmentName, ex);
+                     createIndexSegment(segmentName);
                  } else {
-                     log.warn("Index segment append failed for segment {} due to ", segmentName, ex);
+                     logWarningMessage(segmentName, ex);
```
Comment (tkaitchuck) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java:84]: I don't think we want these two message to be identical, it makes it hard to tell which path was taken when looking at the log.

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java [hunk 2]
```diff
@@ -69,15 +77,42 @@ private void handleIndexAppend(String segmentName) {
              .whenComplete((length, ex) -> {
                  if (ex == null) {
                      log.trace("Index segment append successful for Segment : {} with length {} ", segmentName, length);
+                 } else if (Exceptions.unwrap(ex) instanceof StreamSegmentNotExistsException) {
+                     logWarningMessage(segmentName, ex);
+                     createIndexSegment(segmentName);
```
[the remark this answers is not in the document]
Comment (Bhupender-Y) [reply segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java]: Reverted the changes. Handling this at setupAppend as a part of other PR.

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java (hunk 1 above)
Comment (Bhupender-Y) [reply segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java]: Done.

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexRequestProcessor.java [hunk 3]
```diff
@@ -81,8 +82,8 @@ private static Map.Entry<Long, Long> applySearch(StreamSegmentStore store, Strin
         SegmentProperties properties = store.getStreamSegmentInfo(indexSegmentName, TIMEOUT).join();
         long startIdx = properties.getStartOffset() / ENTRY_SIZE;
         long endIdx = properties.getLength() / ENTRY_SIZE;
-
-        return SortUtils.newtonianSearch(idx -> {
+        //If startIdx and endIdx are same, then pass length of segment as a result.
+        return startIdx == endIdx ? getSegmentLength(store, segment, startIdx) : SortUtils.newtonianSearch(idx -> {
```
Comment (tkaitchuck) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexRequestProcessor.java:86]: As this is a return you can break out the turanary into an if/return and not have any further indentation.

**Review by tkaitchuck — APPROVED** [review]

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexRequestProcessor.java (hunk 3 above)
Comment (Bhupender-Y) [reply segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexRequestProcessor.java]: Done.

**Pull request merged**